### PR TITLE
Run3-gem34 Add a GE21 version with 16 eta partitions

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Add a GE21 version with 16 eta partitions

#### PR validation:

Tested a scenario by visual inspection

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by dildick** [comment]
@bsunanda Can you change https://github.com/cms-sw/cmssw/blob/master/DataFormats/MuonDetId/interface/GEMDetId.h#L33 to ```static constexpr int32_t maxRollId = 16;```? That way we can consider eta partition 0 to be the chamber as in GE1/1 and ME0. Thanks.

**Comment by bsunanda** [comment]
@dildick I am doing that and adding to this PR

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/MuonCommonData/python/testExtendedGeometry2026D54XML_cfi.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        'Geometry/MuonCommonData/data/mb4/2015/v2/mb4.xml',
+        'Geometry/MuonCommonData/data/mb4Shield/2021/v1/mb4Shield.xml',
+        'Geometry/MuonCommonData/data/muonYoke/2021/v2/muonYoke.xml',
+        'Geometry/MuonCommonData/data/csc/2021/v1/csc.xml',
+        'Geometry/MuonCommonData/data/mfshield/2017/v1/mfshield.xml',
+        'Geometry/MuonCommonData/data/mf/2026/v2/mf.xml',
+        'Geometry/MuonCommonData/data/rpcf/2026/v2/rpcf.xml',
+        'Geometry/MuonCommonData/data/gemf/TDR_BaseLine/gemf.xml',
+        'Geometry/MuonCommonData/data/gem11/TDR_BaseLine/gem11.xml',
+        'Geometry/MuonCommonData/data/gem21/TDR_Eta21/gem21.xml',
```
Comment (jshlee) [inline Geometry/MuonCommonData/python/testExtendedGeometry2026D54XML_cfi.py:81]: i think TDR_Eta21 is suppose to be TDR_Eta16

Comment (bsunanda) [reply Geometry/MuonCommonData/python/testExtendedGeometry2026D54XML_cfi.py]: Thanks Jason. Corrected

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by jshlee** [comment]
@bsunanda - I'm seeing roll 17...
region 1 ring 1 station 2 layer 2 chamber 10 roll 17
I think we have a problem in gemDetId

**Comment by jshlee** [comment]
> @bsunanda - I'm seeing roll 17...
> region 1 ring 1 station 2 layer 2 chamber 10 roll 17
> I think we have a problem in gemDetId

@dildick @watson-ij 
I think this is from the trigger. Sven, could you take a look at this.
CSCTriggerPrimitivesProducer/'simCscTriggerPrimitiveDigis'
Exception Message:
GEMDetId ctor: Invalid parameters:  region 1 ring 1 station 2 layer 2 chamber 10 roll 17

**Comment by dildick** [comment]
@jshlee Can you copy the entire exception message? Can't really tell where it is coming from.

**Comment by bsunanda** [comment]
@dildick @jshlee @watson-ij I just checked the DetId code. I could not find it is caused by the DetId code

**Comment by jshlee** [comment]
> @dildick @jshlee @watson-ij I just checked the DetId code. I could not find it is caused by the DetId code

@dildick @bsunanda yep, I suspect it is happening when GEMDetId is constructed somewhere in CSCMotherboardME21GEM.

**Comment by dildick** [comment]
@jshlee So... this geometry will not work with the GE2/1-ME2/1 trigger. The coincidence pad producer may work, but all ILT lookup tables were made for an 8-eta partition geometry. Until that is updated, the code cannot be properly tested. Also the pad producer for GE2/1 needs to be modified.

**Comment by jshlee** [comment]
> @jshlee So... this geometry will not work with the GE2/1-ME2/1 trigger. The coincidence pad producer may work, but all ILT lookup tables were made for an 8-eta partition geometry. Until that is updated, the code cannot be properly tested. Also the pad producer for GE2/1 needs to be modified.

ok, we can turn gem trigger for this geometry to begin with.
it happens in
https://github.com/cms-sw/cmssw/blob/master/L1Trigger/CSCTriggerPrimitives/src/GEMCoPadProcessor.cc#L44

**Comment by dildick** [comment]
In addition to the trigger, also the validation modules will either crash or produce nonsense.

**Comment by dildick** [comment]
And because we have to support the 8 eta partition GE2/1 geometry for the HLT TDR, I think it would be best to enable this geometry only in a specific Phase2 development era.

**Comment by cvuosalo** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**